IDinsight/ask-a-question
# [128] - Internal TTS for AAQ

Reviewer: @lickem22 
Estimate: 30 mins

---

## Ticket

Fixes:[ JIRA_TICKET_LINK](https://idinsight.atlassian.net/browse/AAQ-617?atlOrigin=eyJpIjoiMmJhMzAzYTkxNWYyNGQzZmJlZmE2YTY3ZDVjNjc2ODciLCJwIjoiaiJ9)

## Description
This pull request integrates `Piper` Text-to-Speech (TTS) as an `internal TTS` model for AAQ, supporting both English and Swahili languages.

### Goal
The primary aim of this PR is to incorporate an in-house, open-source TTS model alongside the existing external Google Cloud models. This addition enhances AAQ's speech synthesis capabilities and reduces dependency on third-party services.

### Changes
- Introduced a new environment variable `CUSTOM_TTS_ENDPOINT` to facilitate the utilization of our in-house TTS model services.
- Implemented a new optional TTS model within the `optional_components/speech_api` directory, expanding AAQ's speech synthesis options.
- Added a dedicated Makefile target to create a separate `aaq-speech` environment, streamlining the process of running speech-related tests and working with speech services through a manual setup.

### Future Tasks (optional)
- Integration of` Bhashini TTS` for Indic languages, further expanding AAQ's multilingual capabilities.

## How has this been tested?
Docker compose
Swagger UI
pytests

### How to test this?

1.  Ensure the `CUSTOM_STT_ENDPOINT` and `CUSTOM_TTS_ENDPOINT` environment variables are correctly set in your `.core_backend.env` file.
2. Initialize the Docker containers using the command: `docker compose -f docker-compose.yml -f docker-compose.dev.yml -f docker-compose.speech.yml -p aaq-stack watch.`
3. Utilize the `/voice-search` endpoint and inspect the generated URL for the TTS speech file to confirm proper functionality.

## Checklist

Fill with `x` for completed.
